src/flashbar/flash.tsx

```diff
@@ -110,7 +111,20 @@ export const Flash = React.forwardRef(
     const analyticsMetadata = getAnalyticsMetadataProps(props as BasePropsWithAnalyticsMetadata);
     const elementRef = useComponentMetadata('Flash', PACKAGE_VERSION, { ...analyticsMetadata });
     const mergedRef = useMergeRefs(ref, elementRef);
-    const { discoveredActions, headerRef, contentRef } = useDiscoveredAction(type);
+
+    const { discoveredActions, headerRef: headerRefAction, contentRef: contentRefAction } = useDiscoveredAction(type);
+    const {
+      hasReplacementHeader,
+      hasReplacementContent,
+      headerRef: headerRefContent,
+      contentRef: contentRefContent,
+      replacementHeaderRef,
+      replacementContentRef,
+      actionsRef,
+    } = useDiscoveredContent({ type, header, children: content });
+
+    const headerRef = useMergeRefs(headerRefAction, headerRefContent);
+    const contentRef = useMergeRefs(contentRefAction, contentRefContent);
```

Comment:
We can, but then we'd need to change some API's like useDiscoveredAction so that it would accept a ref instead of giving one. I don't quite like that because ref objects are mutable and passing those around can lead to issues. When each util gives its ref we can expect that ref to be used for that util only.

Reply:
Updated, please check

Reply:
Right, the RefObject is not mutable. You can call it a codestyle then - I do prefer the option with individual refs better. For instance, this also allows to replace a ref object with a callback ref without changes in the consuming component.

In the other PR I replaced one class having a mutable property with another one and the class is only used in tests, I don't see how this is related.